packages/core/src/core/rsbuild.ts

```diff
@@ -145,6 +145,12 @@ export const prepareRsbuild = async (
                 ...(config.module.parser.javascript || {}),
               };
 
+              config.resolve ??= {};
+              // skip `module` field. ESM module resolved by module field is not always a native ESM module
+              config.resolve.mainFields = config.resolve.mainFields?.filter(
+                (filed) => filed !== 'module',
```

Comment:
What if there's a legacy package with `module` and `main`, without `exports` field? like https://github.com/arco-design/arco-design/blob/main/package.json.

Reply:
Updated to: Skip the `module` field only in the `node` test environment and when Node.js version is 18 or below.